lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -1875,7 +1828,9 @@ Scope * ByteCodeGenerator::FindScopeForSym(Scope *symScope, Scope *scope, Js::Pr
 {
     for (scope = scope ? scope->GetEnclosingScope() : currentScope; scope; scope = scope->GetEnclosingScope())
     {
-        if (scope->GetFunc() != funcInfo && scope->GetMustInstantiate() && scope != this->globalScope)
+        if (scope->GetFunc() != funcInfo
+            && (scope->GetMustInstantiate() || (scope->GetScopeType() == ScopeType_Parameter && !scope->GetCanMergeWithBodyScope()))
```

Comment:
Is the param scope special here? Can we just use the must-instantiate flag to indicate that it is not merged and has to be there?